### PR TITLE
CFD-43: Mobile Login/Signup Page

### PR DESCRIPTION
**Pull request by RyanL123**
### Reason for Change

Completes CFD-43

### Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation update

### Testing

Tested by loading into Expo Go

Login page
image

Toast showing error 
image

Password reset code
image

After login
image

### CODE REVIEW

**Comment by RyanL123** [comment]
Not ready for review yet, just want to see whether build succeeds. 

@BakerWJ there's a problem with loading environmental variables in `app.config.ts` for Expo. Since our env variables are in the root directory (the parent directory of Expo), it's not able to load them via `process.env`, but it works if we copy the `.env` file into the `apps/expo`. Interestingly enough Next doesn't care about this, do we have som additional setup? 

Also, `app.config.ts` can read from `process.env` when it's in the context of Github's CI (which is where we're loading `EXPO_PROJECT_ID`), so I'm a little confused on how it's actually loading environmental variables.

File: apps/expo/src/components/ResetPassword.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  clerkError: boolean;
+  errors: {
+    code: string;
+    message: string;
+    longMessage: string;
+    meta: {
+      paramName: string;
+    };
+  }[];
+}
```
Comment (BakerWJ) [inline apps/expo/src/components/ResetPassword.tsx:17]: why is this interface being created manually? can it not be inferred / imported?

File: apps/expo/src/app/index.tsx [hunk 2]
```diff
@@ -27,13 +50,25 @@ const Index = () => {
...
 
   return (
-    <SafeAreaView className="">
-      {/* Changes page title visible on the header */}
-      <Stack.Screen options={{ title: "Home Page" }} />
-      <View className="h-full w-full p-4">
-        <Text className="font-poppins"> Index </Text>
-      </View>
-    </SafeAreaView>
+    <RootSiblingParent>
```
Comment (BakerWJ) [inline apps/expo/src/app/index.tsx:53]: should this be in layout instead?

File: apps/expo/src/app/index.tsx
Comment (BakerWJ) [inline apps/expo/src/app/index.tsx:1]: for now, let's remove this from index.tsx -- it seems to be getting a bit bloated with people's testing

File: apps/expo/src/components/SignUpScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
+import * as React from "react";
```
Comment (BakerWJ) [inline apps/expo/src/components/SignUpScreen.tsx:1]: would recommend not using import *

**Review by BakerWJ — COMMENTED** [review]
looks good -- for now let's remove the signin stuff from index.tsx, but your next ticket will be building out the signin pages

File: apps/expo/src/components/ResetPassword.tsx (hunk 1 above)
Comment (RyanL123) [reply apps/expo/src/components/ResetPassword.tsx:17]: when you catch an error the type is `unknown`, and to be able to access fields on it you must cast it to some known type. Clerk actually doesn't export this, so we have to manually create it

**Review by BakerWJ — APPROVED** [review]
lgtm

**Pull request merged**